src/transformState.js

```diff
@@ -1,11 +1,40 @@
 'use strict';
 
+const ADD = 'addProperties';
+const DELETE = 'removeProperties';
+const CLEAR = 'clear';
+
 /**
  * @param {Object} state
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  actions.forEach(action => {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case ADD:
+        Object.assign(state, extraData);
+
+        return state;
+      case DELETE:
+        keysToRemove.forEach(key => {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
```

Comment:
check can be skipped.
If applied to non-existent property, `delete` just does nothing